src/interpreter.rs

```diff
@@ -12,19 +14,19 @@ use crate::{
 
 pub struct Interpreter {
     pub environment: Rc<RefCell<Environment>>,
-    locals: HashMap<Expr, usize>,
+    locals: IndexMap<Expr, usize>,
```

Comment:
using `IndexMap` means you can have a reasonably stable hash if you need it (because it's an ordered hash map), thats why i added it here